### PR TITLE
Wavefront firehose nozzle that sends PCF metrics to Wavefront proxy l…

### PR DESCRIPTION
**Pull request by sushantdewan123**
Wavefront firehose nozzle that sends PCF metrics to Wavefront proxy locally

### CODE REVIEW

File: src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java
Comment (vikramraman) [inline src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java]: Might make sense to log the exception message here for context when debugging.

File: README.md
Comment (panghy) [inline README.md:7]: Would we not have this prerequisites once we have the tile also deploy the proxy as an app?

File: README.md
Comment (panghy) [inline README.md]: I we should mention that this represents instructions to do this manually?

File: pom.xml
Comment (panghy) [inline pom.xml]: fix alignments in this file

File: src/main/java/com/wavefront/WavefrontNozzleApplication.java
Comment (panghy) [inline src/main/java/com/wavefront/WavefrontNozzleApplication.java:14]: we don't normally do indenting like this, please use the wavefront intellij style

File: src/main/java/com/wavefront/config/BeansConfiguration.java
Comment (panghy) [inline src/main/java/com/wavefront/config/BeansConfiguration.java:31]: blank line

File: src/main/java/com/wavefront/config/BeansConfiguration.java
Comment (panghy) [inline src/main/java/com/wavefront/config/BeansConfiguration.java]: nit-pick, all method invocation chains should line-break after the "." (not before)

File: src/main/java/com/wavefront/props/FirehoseProperties.java
Comment (panghy) [inline src/main/java/com/wavefront/props/FirehoseProperties.java]: use javadoc

File: src/main/java/com/wavefront/proxy/ProxyForwarder.java
Comment (panghy) [inline src/main/java/com/wavefront/proxy/ProxyForwarder.java]: comment or revmoe

File: src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java
Comment (panghy) [inline src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java]: do the toString().length() once

File: src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java
Comment (panghy) [inline src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java]: this should not log every line (do fine with an if isLoggable)

File: src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java
Comment (panghy) [inline src/main/java/com/wavefront/proxy/ProxyForwarderImpl.java]: should also include the exception

File: src/main/java/com/wavefront/utils/ContainerMetricUtils.java
Comment (panghy) [inline src/main/java/com/wavefront/utils/ContainerMetricUtils.java]: comment not really relevant

File: src/main/java/com/wavefront/utils/ContainerMetricUtils.java
Comment (panghy) [inline src/main/java/com/wavefront/utils/ContainerMetricUtils.java]: line too long

File: src/main/java/com/wavefront/utils/MetricUtils.java
Comment (panghy) [inline src/main/java/com/wavefront/utils/MetricUtils.java]: should statically capture this value instead of computing it everytime

File: src/main/java/com/wavefront/utils/MetricUtils.java
Comment (panghy) [inline src/main/java/com/wavefront/utils/MetricUtils.java]: String can mark @Nullable

File: src/test/java/com/wavefront/WavefrontNozzleApplicationTests.java
Comment (panghy) [inline src/test/java/com/wavefront/WavefrontNozzleApplicationTests.java]: Will approve the change when we have some kind of test (just testing whether the spring wirings can be satisfied would be a good start)

**Review by panghy — COMMENTED** [review]
We should decide where @vikramraman's work would live ultimately. It seems to me that there should at least be some clarification as to what this nozzle is used for in the big picture (i.e. this is what the tile installs, blah blah blah, etc.)

File: manifest.yml
Comment (cholick) [inline manifest.yml:4]: `2` is a better default. Never deploy with just 1 instance to avoid downtime (message loss).

**Comment by cholick** [comment]
I could be missing it, not familiar with any of the frameworks in play, but I don't see error handling on issues with the websocket connection. I would expect to see something in `FirehoseToWavefrontProxyConnectorImpl.connect`. The case I'm trying to track down is what happens when your nozzle can't keep up: the websocket is closed with a `1008`. A case worth thinking about since this means the nozzle needs scaling out.

File: src/main/java/com/wavefront/service/FirehoseToWavefrontProxyConnector.java
Comment (panghy) [inline src/main/java/com/wavefront/service/FirehoseToWavefrontProxyConnector.java]: indeting is odd, 47-49 should be one level in (check your ide formatting settings)

File: src/test/java/com/wavefront/WavefrontNozzleApplicationTests.java
Comment (panghy) [inline src/test/java/com/wavefront/WavefrontNozzleApplicationTests.java:37]: indenting is still odd, check to see your IDE setting for this project is the same as the system

**Review by panghy — APPROVED** [review]
LGTM, just some nitpicks

**Review by panghy — APPROVED** [review]

**Pull request merged**
